apps/oxlint/Cargo.toml

```diff
@@ -40,6 +40,7 @@ ignore = { workspace = true, features = ["simd-accel"] }
 insta = { workspace = true }
 miette = { workspace = true }
 rayon = { workspace = true }
+regex = { workspace = true }
```

Comment:
Per https://oxc.rs/docs/contribute/rules.html#performance we should try and avoid using regex as much as possible, most usages can be replaced by simple string replacements.

Also, should this belong to `dev-dependencies`?

Reply:
I forgot about dev-dependencies, moved insta too. 
Simple String replacement is not possible because the output of the diagnostic will change for 

> Finished in \<variable> ms on 1 file with 97 rules using \<variable> threads.